l10n_br_resource/models/resource_calendar.py

```diff
@@ -49,6 +49,9 @@ def _compute_leave_ids(self):
     @api.constrains("parent_id")
     def _check_hierarchy(self):
         if not self._check_recursion():
```

Comment:
@mileo foi incluido um teste para essa constraint, pelo teste ao chamar o metodo _check_recursion já é chamado um UserError o que torna desnecessário o IF e o raise abaixo( você pode verificar tirando o teste do with self.assertRaises("UserError"), por favor veja de confirmar essa questão para podermos simplificar o método, veja a mensagem de erro:

2023-08-22 01:19:32,342 327 ERROR test odoo.addons.l10n_br_resource.tests.test_resource_calendar: ERROR: TestResourceCalendar.test_19_check_constraint_recursive
Traceback (most recent call last):
  File "/odoo/external-src/l10n-brazil-FIX-warning_resource/l10n_br_resource/tests/test_resource_calendar.py", line 343, in test_19_check_constraint_recursive
    self.nacional_calendar_id.write({"parent_id": self.municipal_calendar_id.id})
  File "/odoo/src/odoo/models.py", line 3773, in write
    self.flush_model([self._parent_name])
  File "/odoo/src/odoo/models.py", line 5577, in flush_model
    self._flush(fnames)
  File "/odoo/src/odoo/models.py", line 5650, in _flush
    process(model, id_vals)
  File "/odoo/src/odoo/models.py", line 5600, in process
    model.browse(ids)._write(vals)
  File "/odoo/src/odoo/models.py", line 3856, in _write
    parent_records._parent_store_update()
  File "/odoo/src/odoo/models.py", line 4274, in _parent_store_update
    raise UserError(_("Recursion Detected."))
odoo.exceptions.UserError: Recursion Detected.

Reply:
Como o mantedor do modulo apenas marcou isso como Resolvido considerei que não existe problema na alteração, no último commit foi feita a simplificação do método